Fix build error "variable length array of non-POD element type" when building with clang.
http://forum.simutrans.com/index.php?topic=10477
--- simtypes.h.orig	2012-06-15 16:31:23.000000000 -0500
+++ simtypes.h	2012-10-28 17:29:41.000000000 -0500
@@ -17,6 +17,9 @@
 #	define ALLOCA(type, name, count) type* name = static_cast<type*>(alloca(sizeof(type) * (count)))
 #
 # define inline _inline
+#elif defined __clang__
+#	include <alloca.h>
+#	define ALLOCA(type, name, count) type* name = static_cast<type*>(alloca(sizeof(type) * (count)))
 #else
 #	define ALLOCA(type, name, count) type name[count]
 #endif
